app/javascript/src/NewService/components/ServiceDiscoveryForm.jsx

```diff
@@ -1,47 +1,43 @@
 // @flow
 
-import React from 'react'
-import {useState, useEffect} from 'react'
+import React, {useEffect} from 'react'
 
 import {FormWrapper, ErrorMessage,
   ServiceDiscoveryListItems} from 'NewService/components/FormElements'
 import {fetchData} from 'utilities/utils'
-import type {FormProps} from 'NewService/types'
+import type {ClusterResource} from 'NewService/types'
+import {PROJECTS_PATH} from 'NewService'
 
-const BASE_PATH = '/p/admin/service_discovery'
-const PROJECTS_PATH = `${BASE_PATH}/projects.json`
+type Props = {
+  formActionPath: string,
+  setLoadingProjects: boolean => void
+}
 
-const ServiceDiscoveryForm = ({formActionPath}: {formActionPath: string}) => {
-  const [projects, setProjects] = useState([])
-  const [services, setServices] = useState([])
-  const [fetchErrorMessage, setFetchErrorMessage] = useState('')
+const ServiceDiscoveryForm = ({formActionPath, setLoadingProjects}: Props) => {
+  // Don't use named imports so that useState can be mocked in specs
```

Comment:
The reason why is to inject a spied `setProjects` so that the receiving projects can be asserted. Ideally it would simply assert the projects are rendered but since it's an async flow, there's no way to tell Jest to "wait" except by spying `setProjects`.

I read the issue but I can't see any way to workaround this, it seems all it does is to support hooks functionally when using Enzyme, but not direct access to them.